packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -35,13 +40,63 @@ export const TranslationSidebar = ({
     ['BDB', 'LSJ'].includes(resource)
   );
   const lexiconEntry = lexiconResource?.entry ?? '';
-  const { t } = useTranslation(['common', 'translate']);
+
+  const translatorNotesQuery = useQuery(
+    ['verse-translator-notes', language, verse.id],
+    () => apiClient.verses.findTranslatorNotes(verse.id, language)
+  );
+  const translatorNote = translatorNotesQuery.isSuccess
+    ? translatorNotesQuery.data.data[word.id]
+    : null;
 
   const tabTitles = ['translate:lexicon', 'translate:notes'];
   if (showComments) {
     tabTitles.push('translate:comments');
   }
 
+  const userCan = useAccessControl();
+
+  const canEditNote = userCan('translate', {
+    type: 'Language',
+    id: language,
+  });
+
+  // Only update the content shown in the text editor when the word ID changes.
+  const [originalNoteContent, setOriginalNoteContent] = useState('');
+  const [noteContent, setNoteContent] = useState('');
+  const wordId = useRef('');
+  useEffect(() => {
+    if (word.id !== wordId.current) {
+      wordId.current = word.id;
+      setOriginalNoteContent(translatorNote?.content ?? '');
+      setNoteContent(translatorNote?.content ?? '');
+    }
+  }, [word.id, translatorNote]);
+
+  const saveNote = useCallback(
+    async (noteContent: string) => {
+      await apiClient.words.updateTranslatorNote({
+        wordId: word.id,
+        language,
+        note: noteContent,
+      });
+      translatorNotesQuery.refetch();
+      setNeedsSaved(false);
+    },
+    [language, translatorNotesQuery, word.id]
+  );
+
+  const [needsSaved, setNeedsSaved] = useState(false);
+  // Save the note content after a 500 ms debounce.
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      if (needsSaved) {
+        saveNote(noteContent);
+      }
+    }, 500);
+    return () => clearTimeout(timeoutId);
+  }, [needsSaved, noteContent, saveNote]);
```

Comment:
We should use the [lodash throttle](https://lodash.com/docs/4.17.15#throttle) function in the useCallback. The idea is that the change event can continuously call the saveNote function, but throttle will prevent it from executing too frequently. We actually need throttle and not debounce because debounce won't execute until you stop typing for a bit. We want to save every 15 seconds or so. Using throttle, means that we don't need the useEffect or needsSaved state.
```ts
const saveNote = useCallback(
  throttle(
    () => { /* save note to server */ },
    15000,
    { leading: false, trailing: true }
  ),
  []
)
```

Reply:
This doesn't work quite right. If the note doesn't save instantly when the user blurs the input, they are left questioning if their changes are actually saved correctly. I'll tinker

Reply:
You can use `saveNote.flush()` in the blur handler. Lodash's throttle adds that method to immediately invoke the function.